example-cropping/README.md

```diff
@@ -0,0 +1,779 @@
+```shell
+npx create-react-app appwrite-react
+cd appwrite-react
+```
+While we are in the CLI we will also install the Appwrite JS API by running:
+```shell
+yarn add appwrite
+```
+and finally we will launch the React development server with:
+```shell
+yarn start
+```
+This should launch a server on `localhost:3000` with Live Reload.
+
+## Introducing the Appwrite SDK
+With the boilerplate now complete we can now initialise the Appwrite SDK in the project before working on the cropping tutorial. To keep things clean we will initialise this in it's own file, we will create this file in `src/` and call it `utils.js`. Within this file go ahead and paste the following code:
+```js
+import 'appwrite';
+const appwrite = new window.Appwrite();
+function appWrite({ endpoint, projectId }) {
+  let ep = endpoint || 'http://localhost/v1';
+  let pi = projectId || 'ProjectID';
+  return appwrite
+    .setEndpoint(ep)
+    .setProject(pi)
+
+}
+export { appWrite };
+
+```
+
+
+## How To Use Cropping Using Appwrite SDK
+
+To crop an image using appwrite SDK is actually pretty simple you just need to upload your file into appwrite storage then preview the image and pass the optional parameters, if you take a look at the js docs in https://appwrite.io/docs/client/storage#getFilePreview you will see all of the optional parameters and how to use it. Below is the example of how to use it using js SDK
+
+```js
+let sdk = new Appwrite();
+
+sdk
+    .setEndpoint('https://[HOSTNAME_OR_IP]/v1') // Your API Endpoint
+    .setProject('5df5acd0d48c2') // Your project ID
+;
+
+let result = sdk.storage.getFilePreview('[FILE_ID]',300,null,100,null,"webp");
+
+console.log(result); // Resource URL for example http://localhost/v1/storage/files/5f70694f8637d/preview?width=300&quality=100&output=webp&project=5f6f49028c5b9
+
+```
+
+using the above script we will get our image with the output that we like so for example we pass 300 as width,100 as quality and output webp we will get exactly the output like that without the need of cdn to change the width, height, quality ,background, and output the appwrite will do it just like we want. Now we will create a component that can implement this functionality using react, below is step by step how to create simple app that can crop our image from appwrite storage starting from signup, login, upload, list, then preview and crop.
+
+## Step 1 - Create PreviewImage Component
+
+So this is actually the main ingredient in our tutorial what this does is to show the preview of the image in our server and return in a spesific parameter that we want so in order to use this component we can pass the parameter exactly the same as in the appwrite SDK documentation which is id,width,height,quality,background,and output and it will output the image with that spesific condition which mean this component is used for cropping an image that we get from our storage.
```

Comment:
```suggestion
So this is actually the main ingredient in our tutorial what this does is to show the preview of the image in our server and return in a spesific parameter that we want so in order to use this component we can pass the parameter exactly the same as in the appwrite SDK documentation which is id, width, height, quality, background and output and it will output the image with that specific condition which means this component is used for cropping an image that we get from our storage.
```